src/main/kotlin/racingcar/domain/car/Car.kt

```diff
@@ -1,29 +1,35 @@
 package racingcar.domain.car
 
+import racingcar.domain.car.carengine.CarEngine
+import racingcar.domain.car.carengine.RandomCarEngine
 import racingcar.domain.record.CarRecord
 
 class Car(
-    val lineNumber: Int,
-    private val carEngine: CarEngine = RandomCarEngine
+    val name: String,
+    private val carEngine: CarEngine = RandomCarEngine,
+    position: Int = START_POSITION
 ) {
-    var position: Int = START_POSITION
+    var position: Int = position
         private set
+    var carRecord = CarRecord(listOf())
+        private set
+
+    init {
+        require(name.length <= 5) { NAME_LIMIT_ERROR_MESSAGE }
+    }
 
     fun move() {
         if (carEngine.canGo()) position++
+        carRecord.record += position
     }
 
-    fun race(tryCount: Int): CarRecord {
-        val record = arrayListOf<Int>()
-        repeat(tryCount) {
-            move()
-            record.add(this.position)
-        }
-
-        return CarRecord(this, record)
+    fun race(tryCount: Int) {
+        repeat(tryCount) { move() }
     }
 
     companion object {
         const val START_POSITION = 0
+        const val NAME_LENGTH_LIMIT = 5
```

Comment:
```suggestion
        private const val START_POSITION = 0
        private const val NAME_LENGTH_LIMIT = 5
```
내부에서만 접근한다면, private 접근자를 붙여보면 어떨까요?

Reply:
테스트 코드에서 사용되는 상수를 제외하고 private 접근자를 사용하도록 했습니다.